bindings/matrix-sdk-ffi/src/authentication_service.rs

```diff
@@ -116,6 +118,10 @@ impl AuthenticationService {
         let client = builder.build().map_err(AuthenticationError::from)?;
 
         RUNTIME.block_on(async move {
+            if client.sliding_sync_proxy().await.is_none() {
+                return Err(AuthenticationError::SlidingSyncNotAvailable);
+            }
```

Comment:
Maybe controversial that you can't authenticate against a server that doesn't support sliding sync, although it should still be possible by building a client through a `ClientBuilder` and not using this service.